lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -236,6 +236,8 @@ def disallows_original_or_typecast_value?(value)
         def disallowed_values
           if collection?
             [Array.new]
+          elsif attachment?
```

Comment:
Instead of adding a new clause here, what if you were to add another condition to `attribute_accepts_string_values`?

Reply:
you mean a `&& !attachment?` there? I agree that a `if` condition here is bad, but `attribute_accepts_string_values` is also quite complex already. IMO, I'd just move complexity and not make readable. I'm also worried about future side-effects... maybe we can do early returns instead of `if..else` statements, like:

```
return [Array.new] if collection?
return [nil] if attachment?

(...)
```

I think it would make the method more readable.

Reply:
Hmm, I'm not a fan of early returns. I think it looks pretty, but at the same time it makes the logic more difficult to untangle and reason about later. With `if`-`else`s I find I can mentally draw a flowchart in my head a lot more quickly, even if it looks "uglier".

This is probably fine for now, until we have other requirements.